src/NUnitTestAdapter/NUnitTestFilterBuilder.cs

```diff
@@ -29,15 +29,22 @@ public TestFilter ConvertTfsFilterToNUnitFilter(TfsTestFilter tfsFilter, List<Te
             return MakeTestFilter(testCases);
         }
 
-        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases)
+        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases = null, string where = null)
```

Comment:
Use of nullable parameters here seems to make this method having too many executable paths, and it is unclear what they do.  I would prefer this to be explicitly coded.  Right now I don't fully understand what the different options will be here.

Reply:
This was due to me trying to keep the footprint of the PR low. I can add an explicit method and update the call site.

Reply:
I split off the new functionality and renamed the existing method to be clearer. It is much more straight forward now I think ^_^